packages/expo-device/ios/EXDevice/EXDevice.m

```diff
@@ -0,0 +1,316 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXDevice/EXDevice.h>
+#import <EXDevice/EXDeviceUID.h>
+#import <UMCore/UMUtilities.h>
+
+#import <ifaddrs.h>
+#import <arpa/inet.h>
+#import <mach-o/arch.h>
+#import <CoreLocation/CoreLocation.h>
+#import <UIKit/UIKit.h>
+#import <sys/utsname.h>
+
+#import <CoreTelephony/CTCallCenter.h>
+#import <CoreTelephony/CTCall.h>
+#import <CoreTelephony/CTCarrier.h>
+#import <CoreTelephony/CTTelephonyNetworkInfo.h>
+
+#import <UMCore/UMUtilitiesInterface.h>
+#import <UMCore/UMUtilities.h>
+
+#import <WebKit/WKWebView.h>
+
+#if !(TARGET_OS_TV)
+#import <LocalAuthentication/LocalAuthentication.h>
+@import Darwin.sys.sysctl;
+#endif
+
+@interface EXDevice() {
+}
+
+@property (nonatomic, strong) NSString *webViewUserAgent;
+@property (nonatomic) bool isEmulator;
+
+@end
+
+@implementation EXDevice
+
+UM_EXPORT_MODULE(ExpoDevice);
+
+- (dispatch_queue_t)methodQueue
+{
+  return dispatch_get_main_queue();
+}
+
+
+UM_EXPORT_METHOD_AS(getUserAgentAsync,
+                    getWebViewUserAgentWithResolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  __weak EXDevice *weakSelf = self;
+  EXDevice *strongSelf = weakSelf;
+  
+  __block WKWebView *webView;
```

Comment:
@sjchmiela but isn't `strongSelf` an undeclared identifier in the condition of the the `if` since it isn't defined until we're inside the `if` statement?

Reply:
unless if change ` if (!strongSelf.webViewUserAgent)` to ` if (_webViewUserAgent)`